src/openrct2/interface/Viewport.cpp

```diff
@@ -952,7 +952,7 @@ void viewport_paint(
     if (recorded_sessions != nullptr)
     {
         const uint16_t columnSize = rightBorder - alignedX;
-        const uint16_t columnCount = (columnSize / 32);
+        const uint16_t columnCount = (columnSize / 32) + (columnSize % 32 == 0 ? 0 : 1);
```

Comment:
The problematic scenario is when `columCount == 0`, is it not? This change is making `columnCount = (columnSize / 32) + 1` for every `columnSize` above `32` (excluding multiples of 32), along with smaller than `32`. Why not just:
```suggestion
        const uint16_t columnCount = std::max((columnSize / 32), 1);
```

Reply:
I don't quite follow. It has to only add +1 when there is a remainder, ex 33. If its 0 then it should stay at 0.

Reply:
@tupaschoal If you divide integers in C, it will always floor them. So `31 / 32` = 0. What you propose might work if you convert the lot to float, but that comes with its own set of nasties.

The code seems correct, the only thing that I would change is stick a comment above saying `Divide and round up` or something like that.

Reply:
@Gymnasiast I don't think that's what he's proposing at all? `std::max` would work with the same integer division you've outlined.

Reply:
@AaronVanGeffen It won't produce the same results as those of ZehMatt. Take a `columnSize` of 48 for example - Matt's code will return 2, while Tulio's will return 1.

Reply:
I was assuming it was crashing only when `columnCount` ended up being 0, hence the max proposition that would always return at least 1.

Reply:
If you want to round up why not do: `(columnSize + 31) / 32`?

Reply:
Yup, @IntelOrca's is the canonical version of ceiling division.